### PR TITLE
SYS-3338 Adds sudo extrinsic that sanistises the external reference

### PR DESCRIPTION
**Pull request by thadouk**
## Proposed changes
Ensures that nfts external references is within the expected bounds.

## Type of change/Merge

🚨What type of change is this PR? </br>
_Put an `x` in the boxes that apply_

- [ ] Release 
  - [ ] Increase versions 
  - [ ] Baseline tests passed  
  - Release type:
    - [ ] Major release 
    - [ ] Minor release 
    - [ ] Patch release 

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._

- [ ] You describe the purpose of the PR, e.g.:
  - What does it do?
  - Highlight what important points reviewers should know about;
  - Indicates if there is something left for follow-up PRs.
- [ ] Documentation updated
- [ ] Business logic tested successfully
- [ ] Verify First, Write Last: In Substrate development, it is important that you always ensure preconditions are met and return errors at the beginning. After these checks have completed, then you may begin the function's computation.

### CODE REVIEW

**Review by ivan-cholakov — APPROVED** [review]

File: pallets/nft-manager/src/lib.rs [hunk 1]
```diff
@@ -774,6 +774,35 @@ pub mod pallet {
...
+
+        #[pallet::weight(T::DbWeight::get().reads_writes(1 as u64, 2 as u64))]
+        #[pallet::call_index(11)]
+        pub fn sanitise_nft_data(origin: OriginFor<T>, nft_ids: Vec<U256>) -> DispatchResult {
+            let _sender = ensure_root(origin)?;
+            let bound: usize = <NftExternalRefBound as sp_core::Get<u32>>::get() as usize;
+            for nft_id in nft_ids.iter() {
+                if let Some(nft) = Self::nfts(nft_id) {
+                    if nft.unique_external_ref.len() <= bound {
+                        log::info!("NFT with id: {} was not found", nft_id);
```
Comment (nahuseyoum) [inline pallets/nft-manager/src/lib.rs:786]: Should we update the error message. The nft is found but the external ref length is not over the bound.

**Review by nahuseyoum — APPROVED** [review]

File: pallets/nft-manager/src/lib.rs (hunk 1 above)
Comment (thadouk) [reply pallets/nft-manager/src/lib.rs]: Yeah Alex raised that as well. It has been updated on the latest version

File: pallets/nft-manager/src/lib.rs [hunk 2]
```diff
@@ -774,6 +774,35 @@ pub mod pallet {
 
             Ok(())
         }
+
+        #[pallet::weight(T::DbWeight::get().reads_writes(1 as u64, 2 as u64))]
+        #[pallet::call_index(11)]
+        pub fn sanitise_nft_data(origin: OriginFor<T>, nft_ids: Vec<U256>) -> DispatchResult {
```
[the remark this answers is not in the document]
Comment (thadouk) [reply pallets/nft-manager/src/lib.rs]: done

**Review by alexpArtos — APPROVED** [review]

File: pallets/nft-manager/src/lib.rs [hunk 3]
```diff
@@ -774,6 +774,35 @@ pub mod pallet {
...
+                        log::info!("NFT with id: {} was not found", nft_id);
+                        continue
+                    }
+                    let (original_ref, sanitised_ref) =
+                        Self::get_sanitised_reference_data(nft, bound);
+                    ensure!(
+                        UsedExternalReferences::<T>::contains_key(&sanitised_ref) == false,
+                        Error::<T>::ExternalRefIsAlreadyInUse
+                    );
+                    let value = UsedExternalReferences::<T>::take(&original_ref);
```
[the remark this answers is not in the document]
Comment (alexpArtos) [reply pallets/nft-manager/src/lib.rs:795]: We (with @thadouk ) checked the source code and it appears that `storageMap.take` calls `unhashed::take` which does remove the key, so we're good.

**Pull request merged**